source/publish/lizmap_plugin/editing.rst

```diff
@@ -155,3 +155,65 @@ This allow to store the sent media files and images in a centralized folder, for
   - media
 
     * tree_pictures/ <-- to store images in this folder, use: :file:`media/tree_pictures/`
+
+Use a WebDAV remote storage
+_________________________________
+
+It is possible to store your files in a remote server that supports the `WebDAV protocol` (e.g. Nextcloud). To achieve this you must:
+
+- Configure the :guilabel:`Attachments` edit widget for the choosen field as follow:
+
+  1. In the :guilabel:`Storage type` field, select the `WebDAV Storage` option
+  2. In the :guilabel:`External Storage`` section, on the :guilabel:`Store URL` field, add an :code:`expression` indicates the full url of the remote path for the file. The expression string must:
+
+    * start with the remote storage url of the :code:`root` folder of the WebDAV server
+    * include or not any destination subfolder 
+    * end with the :code:`file_name(@selected_file_path)` expression
+
+  3. update the :guilabel:`Authentication` section with your :code:`user name` and :code:`password` (optional)
+
+  ..  image:: /images/webdav-configure-attachment-widget.jpg
+    :align: center
+
+  For example, if you want to upload a file in your WebDav server :code:`https://webdav/dav` in the subfolder :code:`pictures/2023/` then the full expression will be:
+  
+  :code:`'https://webdav/dav/pictures/2023/'||file_name(@selected_file_path)`
+
+  You can also upload a file directly in the root folder, simply change the expression above in :code:`'https://webdav/dav/'||file_name(@selected_file_path)`
+
+  .. tip:: You can configure different fields on the same layer or in different layers to upload files to your remote storage under different subfolders.
+
+- Configure your Lizmap installation to recognize your WebDAV storage. To do so in your :code:`var/config/profiles.ini.php` edit the :code:`[webdav:default]` section:
+
+  .. code-block:: ini
+
+    [webdav:default]
+    baseUri=https://webdav/dav/
+    enabled=1
+    user=mywebdavuser
+    password=mywebdavpassword
+
+  where:
+
+    * :code:`baseUri`  is the root folder of yuor webdav server. It must end with ``/`` and it must be the same as indicated in the Attachment widget
+    * :code:`enabled`  set it to 1 switch on the webdav configuration, or to 0 to switch it off
+    * :code:`user`     webdav baseUri authentication user, same as same as configured in the Authentication section of External storage configuration (Attachment widget)
+    * :code:`password` webdav baseUri authentication password, same as configured in the Authentication section of External storage configuration (Attachment widget)
+
+  .. note:: :code:`user` and :code:`password` fields are not mandatory, depends on how you intend to manage the webdav storage.
+
+  .. note:: Configure multiple webdav servers on the same Lizmap installation is not supported, so you can configure only one webdav endpoint per time.     
+
+  .. warning:: To get files from remote storage the :code:`baseUri` will be exposed on the web client.
```

Comment:
```suggestion
  .. warning::
     To get files, outside of the Lizmap Web Client interface, for instance from QGIS Desktop, the :code:`baseUri` must be exposed on the network.
```

Sorry I'm not sure if I'm correct. Feel free to skip.
It's not required to expose the public webdav URL if I'm correct ? (only if you plan to use from QGIS desktop for instance)

Reply:
Sorry, I probably was not clear on this part. 
Since the URL replacement is performed on client side (this replacement is needed to pass correct parameter to the `getMedia` service to get the file), the webdav baseUri is exposed on the browser (not directly of course, but it can be browsed in console). This warning is added also in the [profiles.ini.php.dist](https://github.com/3liz/lizmap-web-client/blob/58f5235efa56df1286e3255b509323e2fc4cfb03/lizmap/var/config/profiles.ini.php.dist#L165), here I just wanted to remark this fact, without specifying how this url is exposed. 

If there is any other information you think is necessary about this part I will add it

Reply:
Ok, sorry I forgot that the replacement was done on the client side. I know it has been a long discussion, and the Christmas holiday was in between now ...